src/settings.cpp

```diff
@@ -280,52 +275,35 @@ static int ParseIntList(const char *p, T *items, size_t maxitems)
 
 	/* If we have read comma but no number after it, fail.
 	 * We have read comma when (n != 0) and comma is not allowed */
-	if (n != 0 && !comma) return -1;
+	if (!result.empty() && !comma) return std::nullopt;
 
-	return ClampTo<int>(n);
+	return result;
 }
 
 /**
  * Load parsed string-values into an integer-array (intlist)
  * @param str the string that contains the values (and will be parsed)
  * @param array pointer to the integer-arrays that will be filled
- * @param nelems the number of elements the array holds. Maximum is 64 elements
+ * @param nelems the number of elements the array holds.
  * @param type the type of elements the array holds (eg INT8, UINT16, etc.)
  * @return return true on success and false on error
  */
 static bool LoadIntList(const char *str, void *array, int nelems, VarType type)
```

Comment:
how easy would it be to get rid of the void* out param and casting to char* at the same time?

Reply:
The `void*` comes from `GetVariableAddress`, which you already pass a `void*` to so arguably that would need to be changed as well which basically entails rewriting the whole variable access bit of the save-load system.

I would expect that touches thousands if not tens of thousands lines of code. It'd probably be worth it in the long run, as we can get rid of defining the types of the variables but alas... a hell of a lot of work.